### DISCUSSION TITLE
Noise calculation of Raman spectra

### DISCUSSION THREAD

**Discussion by georgievgeorgi** [h2020charisma/ramanchada2#79] [Ideas]
@nicocopez, @lellid

I am trying to automate the peak finding algorithm. In order to do so a reliable estimate for the noise is needed. [Here](https://github.com/h2020charisma/ramanchada2/blob/6759d637954e6db951b47dc2cca45d47888c49f8/src/ramanchada2/spectrum/spectrum.py#L142) I implemented the algorithm proposed by @lellid but the results, however, are still not satisfactory.

The idea behind the algorithm is to subtract a smooth version of the signal (thinking that smoothing will suppress the noise) from the original signal. This should result to noise-only spectrum. The signal smoothing is done by savgol(width=3, order=1). Using wider window will be averaging also the peaks.

Here is an example that illustrates the problem:
```python
import ramanchada2 as rc2

ps_spe = rc2.spectrum.from_test_spe(laser_wl=['785'], sample=['PST'], OP=['03'])
ps_ax = ps_spe.subtract_moving_minimum(1000).plot()
ps_ax.axhline(ps_spe.y_noise, c='r')
ps_ax.set_ylim(-ps_spe.y_noise, ps_spe.y_noise * 10)
ps_ax.set_title('Polystyrene')

ne_spe = rc2.spectrum.from_test_spe(laser_wl=['785'], sample=['Neon'], OP=['03'])
ne_ax = ne_spe.subtract_moving_minimum(1000).plot()
ne_ax.axhline(ne_spe.y_noise, c='r')
ne_ax.set_ylim(-ne_spe.y_noise, ne_spe.y_noise * 10)
ne_ax.set_title('Neon')
```
In the case of Neon the noise is a little overestimated due to the sharp peaks. While in the polystyrene case the noise underestimated. Possible reason for this is that the "noise" in polystyrene measurement does not have that high frequency.

For example here:
```python
ax = ps_spe.plot(fmt=':.')
ax.set_xlim(2000, 2100)
ax.set_ylim(1020, 1040)
```

**Comment by nicocopez** [comment]
Sorry, even though I plan to allocated some specific time to play with your code and try a couple of things, I did not have time yet to look into it. 

In short, I was thinking in maybe using SNIP with a large radious of the rolling ball to substract the whole spectrum. So the same approach as using savitzky golay.

Alternatively, after baseline substraction, cut all the higher intensities from the values in the spectrum (removing the peaks) and see what is left, if we end up with the background, this could give a value of the noise.

But i am not sure how well this will work.
